engine/src/test/java/org/camunda/bpm/engine/test/api/mgmt/telemetry/TelemetryMultipleEnginesTest.java

```diff
@@ -0,0 +1,213 @@
+
+    defaultMetricsRegistry.markOccurrence(Metrics.EXECUTED_DECISION_INSTANCES);
+    secondMetricsRegistry.markOccurrence(Metrics.ROOT_PROCESS_INSTANCE_START);
+
+    persistMetrics();
+
+    // when
+    defaultTelemetryReporter.reportNow();
+    secondTelemetryReporter.reportNow();
+
+    // then
+    List<LoggedRequest> requests = wireMockRule.findAll(postRequestedFor(urlEqualTo(TELEMETRY_ENDPOINT_PATH)));
+
+    assertThat(requests).hasSize(2);
+
+    Gson gson = new Gson();
+
+    LoggedRequest defaultRequest = requests.get(0);
+    Data defaultRequestBody = gson.fromJson(defaultRequest.getBodyAsString(), Data.class);
+    assertReportedMetrics(defaultRequestBody, 0, 1, 0, 0);
+
+    LoggedRequest secondRequest = requests.get(1);
+    Data secondRequestBody = gson.fromJson(secondRequest.getBodyAsString(), Data.class);
+    assertReportedMetrics(secondRequestBody, 1, 0, 0, 0);
+  }
+
+  private TelemetryReporter getTelemetryReporter(ProcessEngine engine) {
+
+    ProcessEngineConfigurationImpl engineConfiguration = (ProcessEngineConfigurationImpl)
+        engine.getProcessEngineConfiguration();
+    return engineConfiguration.getTelemetryReporter();
+  }
+
+  private MetricsRegistry getMetricsRegistry(ProcessEngine engine) {
+    ProcessEngineConfigurationImpl processEngineConfiguration = (ProcessEngineConfigurationImpl) engine.getProcessEngineConfiguration();
+    return processEngineConfiguration.getMetricsRegistry();
+  }
+
+  private void assertReportedMetrics(
+      Data data,
+      int expectedRootInstances,
+      int expectedDecisionInstances,
+      int expectedFlowNodeInstances,
+      int expectedTaskWorkers) {
+    Internals internals = data.getProduct().getInternals();
+
+    assertMetric(internals, Metrics.ROOT_PROCESS_INSTANCE_START, expectedRootInstances);
+    assertMetric(internals, Metrics.EXECUTED_DECISION_INSTANCES, expectedDecisionInstances);
+    assertMetric(internals, Metrics.ACTIVTY_INSTANCE_START, expectedFlowNodeInstances);
+    assertMetric(internals, "unique-task-workers", expectedTaskWorkers);
+  }
+
+  private void assertMetric(Internals internals, String name, int expectedCount) {
+
+    Map<String, Metric> metrics = internals.getMetrics();
+
+    Metric rootMetric = metrics.get(name);
+    assertThat(rootMetric).isNotNull();
+    assertThat(rootMetric.getCount()).describedAs("metric " + name).isEqualTo(expectedCount);
+  }
+
+  private void persistMetrics() {
+    persistMetrics(defaultEngine);
+    persistMetrics(secondEngine);
+  }
+
+  private void persistMetrics(ProcessEngine engine) {
+    ProcessEngineConfigurationImpl processEngineConfiguration = (ProcessEngineConfigurationImpl) engine.getProcessEngineConfiguration();
+    DbMetricsReporter metricsReporter = processEngineConfiguration.getDbMetricsReporter();
+    metricsReporter.reportNow();
+  }
+
+  @After
+  public void tearDown() {
+    WireMock.resetAllRequests();
+    clearMetrics();
+  }
+
+  protected void clearMetrics() {
```

Comment:
The metrics are cleaned up only for default engine.

Reply:
It doesn't make a difference right now, but it's probably nicer to write the test cleanly from the beginning.